### PR TITLE
ci: add lint workflow

### LINKED ISSUE #5: ci: lint workflow
breaking #1 into two issues

add a workflow that runs all the linters on every push.

### PR DESCRIPTION
**Pull request by backwardspy**
closes #5

### CODE REVIEW

**Review by sgoudham — CHANGES_REQUESTED** [review]
Just a few comments! Overall, great work!

File: .github/workflows/lint.yaml [hunk 1]
```diff
@@ -0,0 +1,23 @@
+name: Lint
+on: push
```
[the remark this answers is not in the document]
Comment (backwardspy) [reply .github/workflows/lint.yaml:2]: good idea, will do that!

**Comment by backwardspy** [comment]
@sgoudham done, seems to work as no run happened when i pushed that change, but then it did run when i squashed & force pushed. :)

**Review by sgoudham — APPROVED** [review]
Nice use of the `**.py` syntax instead of `**/*.py`!
LGTM :D

**Pull request merged**